### PR TITLE
Traps_Interrupts_Exceptions.rst: update chapter

### PR DESCRIPTION
**Pull request by ASintzoff**

### CODE REVIEW

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 1]
```diff
@@ -1,25 +1,126 @@
...
+.. _cva6_traps_interrupts_exceptions:
+
+Traps, Interrupts, Exceptions
+=============================
+Traps are composed of interrupts and exceptions.
+Interrupts are asynchronous events whereas exceptions are synchronous ones.
+On one hand, interrupts are occuring independently of the instructions
+(mainly raised by peripherals or debug module).
+On the other hand, an instruction may raise exceptions synchronously.
+
```
Comment (jquevremont) [inline docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:28]: Suggest to mention that U-mode delegation of interrupts and exceptions is not supported.

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 2]
```diff
@@ -1,25 +1,126 @@
...
+* breakpoint
+* load address misaligned:
+
+  * ``LH`` at 2n+1 address
+  * ``LW`` at 4n+1, 4n+2, 4n+3 address
+
+* load access fault
+
+  * access to PMP region without read permissions
```
Comment (jquevremont) [inline docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:76]: What about write and execute permissions?

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 3]
```diff
@@ -1,25 +1,126 @@
...
+
+* store/AMO address misaligned
+
+  * ``SH`` at 2n+1 address
+  * ``SW`` at 4n+1, 4n+2, 4n+3 address
+
+* store/AMO access fault
+
+  * access to PMP region without write permissions
```
Comment (jquevremont) [inline docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:85]: What about write and execute permissions?

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 4]
```diff
@@ -1,25 +1,126 @@
...
+
+* environment call from U-mode
+* environment call from S-mode
+* environment call from M-mode
```
Comment (jquevremont) [inline docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:89]: Maybe you could be more specific by mentioning the ECALL instruction. Do you mean EBREAK too?

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 5]
```diff
@@ -1,25 +1,126 @@
...
+* instruction page fault
+* load page fault
+
+  * access to effective address without read permissions
+
+* store/AMO page fault
+
+  * access to effective address without write permissions
+
+* debug request (custom)
```
Comment (jquevremont) [inline docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:99]: Could you specify the source: debug interface, EBREAK instruction?

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 6]
```diff
@@ -1,25 +1,126 @@
...
+
+Note: all exceptions are supported except the ones linked to the hypervisor extension.
```
Comment (jquevremont) [inline docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:101]: I suggest to anticipate the merging of the hypervisor extension, by adding this sentence: "The hypervisor extension is a foreseen augmentation of CVA6 that is not documented in this guide."

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 7]
```diff
@@ -1,25 +1,126 @@
...
+
+Trap return
+===========
+Trap handler ends with trap return instruction (MRET, SRET). The behaviour of the CVA6 core depends on several CSRs.
+
+Configuration CSRs
+------------------
+* ``mstatus``: several fields are read like previous privilege mode (MPP, SPP), previous interrupt enabled (MPIE, SPIE)
```
Comment (jquevremont) [inline docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:109]: Same remark: Who / what entity reads?

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 8]
```diff
@@ -1,25 +1,126 @@
...
+CSRs having an effect on the core behaviour when a trap occurs are:
+
+* ``mstatus`` and ``sstatus``: several fields are read like interrupt enable (MIE, SIE)
+* ``mtvec`` and ``stvec``: specifies the address of trap handler.
+* ``medeleg``: specifies which exceptions can be handled by a lower privileged mode
+* ``mideleg``: specifies which interrupts can be handled by a lower privileged mode
+
+Modified CSRs
+-------------
+* ``mstatus`` or ``sstatus``: several fields are updated like previous privilege mode (MPP, SPP), previous interrupt enabled (MPIE, SPIE)
```
Comment (jquevremont) [inline docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:46]: Updated by the core?

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst (hunk 1 above)
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:28]: AFAIK, RISC-V privilege spec does not support (even optionally) U-mode delegation. So, I'm not sure it is useful to mention that in our manual.

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 9]
```diff
@@ -1,25 +1,126 @@
...
+=============
+When a trap is raised, the behaviour of the CVA6 core depends on
+several CSRs and some CSRs are modified. The CSR description is available
+in :doc:`CV32A6_Control_Status_Registers.rst`.
+
+Configuration CSRs
+------------------
+CSRs having an effect on the core behaviour when a trap occurs are:
+
+* ``mstatus`` and ``sstatus``: several fields are read like interrupt enable (MIE, SIE)
```
[the remark this answers is not in the document]
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:39]: Rephrased

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 10]
```diff
@@ -1,25 +1,126 @@
...
+--------------------
+The following exceptions are supported by the CVA6:
+
+* instruction address misaligned
+
+  * control flow instruction with misaligned target
+
+* instruction access fault
+
+  * physical address fault
```
[the remark this answers is not in the document]
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:61]: Rephrased with PMP permissions

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst (hunk 2 above)
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:76]: For load, only read permissions

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst (hunk 3 above)
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:85]: For store, only write permissions

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst (hunk 4 above)
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:89]: Environment call is only ECALL.

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst (hunk 5 above)
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:99]: Added

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst (hunk 6 above)
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:101]: I prefer to not anticipate.

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst (hunk 8 above)
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:46]: Sentence added.

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 11]
```diff
@@ -1,25 +1,126 @@
...
+
+Configuration CSRs
+------------------
+* ``mstatus``: several fields are read like previous privilege mode (MPP, SPP), previous interrupt enabled (MPIE, SPIE)
+
+Modified CSRs
+-------------
+* ``mstatus``: several fields are updated like interrupt enable (MIE, SIE), modify privilege (MPRV)
+
+Interrupts
```
[the remark this answers is not in the document]
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:115]: Interrupts are already introduced at the beginning of the chapter.

File: docs/01_cva6_user/Traps_Interrupts_Exceptions.rst [hunk 12]
```diff
@@ -1,25 +1,126 @@
...
+==========
+* external interrupt: ``irq_i`` signal
+* software interrupt (inter-processor interrupt): ``ipi_i`` signal
+* timer interrupt: ``time_irq_i`` signal
+* debug interrupt: ``debug_req_i`` signal
```
[the remark this answers is not in the document]
Comment (ASintzoff) [reply docs/01_cva6_user/Traps_Interrupts_Exceptions.rst:120]: Thanks to exception code field in `mcause` CSR. Added in the document.

**Pull request merged**